### PR TITLE
test(rs-sdk): replace print-based DPNS tests with assertions, add identity error paths

### PR DESCRIPTION
**Pull request by thepastaclaw**
## Summary

Replace print-based "visual inspection" tests in `rs-sdk` with real assertions, add error path tests for identity fetches, and document magic values.

Addresses audit findings C01, I02, I03 from the SDK test quality audit.

### C01: DPNS tests — println → assertions

**`dpns_unit_tests.rs`** (3 unit tests):
- `test_dpns_validation_functions`: Replace emoji/println with `assert_eq!` for username validation, homograph conversion, and contested name checks. Expected values were already present in the code but never compared programmatically.
- `test_dpns_edge_cases`: Replace println with `assert!`/`assert!(!...)` for length boundaries, special characters, and unicode rejection.
- `test_dpns_homograph_safety`: Replace println with `assert_eq!` for homograph conversions and idempotency checks.

**`dpns_queries_test.rs`** (2 integration tests, `#[ignore]`):
- `test_dpns_queries_from_docs`: Replace match/println error swallowing with `.expect()` — network errors now properly fail the test.
- `test_dpns_search_variations`: Same treatment.

### I02/I03: Identity fetch assertions + error paths

**`identity.rs`**:
- `test_identity_balance_read`: Add `assert!(balance > 0)`.
- `test_identity_balance_revision_read`: Add `assert!(balance > 0)` and `assert!(revision >= 1)`.
- New: `test_identity_read_not_found` — fetch with `UNKNOWN_IDENTITY_ID`, assert `Ok(None)`.
- New: `test_identity_balance_read_not_found` — fetch balance with `UNKNOWN_IDENTITY_ID`, assert `Ok(None)`.

### Magic value documentation

- `identity.rs`: Document `UNIQUE_KEY_HASH` source (genesis state creation).
- `dpns_queries_test.rs`: Document `TEST_IDENTITY_ID`, `TEST_USERNAME`, `TEST_PREFIX` sources and testnet node address.
- `generated_data.rs`: Document `UNKNOWN_TOKEN_ID` collision with `IDENTITY_ID_1` and why it's acceptable (different namespaces).

## Verification

- `cargo check -p dash-sdk --features=mocks --tests` passes cleanly (only pre-existing `dead_code` warnings).
- Integration tests are `#[ignore]` (require running testnet node).

## Summary by CodeRabbit

* **Tests**
  * Quieter, stronger test suite with clearer assertions, broader edge-case coverage, and added identity/identity-balance “not found” checks; new test harnesses for resolver and symbol validation.

* **Build / Tooling**
  * Improved iOS build tooling: robust static-lib resolution and enhanced simulator/device library assembly with clearer diagnostics for missing libraries or symbols.

* **Documentation**
  * Clarified test-data notes about a non-existent token identifier and namespace distinctions in test state.

## Validation

1. **What was tested**
   - CI checks on this PR (Rust packages linting, tests, builds)

2. **Results**
   - All Rust CI checks passing

3. **Environment**
   - GitHub Actions CI for `dashpay/platform`

### CODE REVIEW

**Comment by thepastaclaw** [comment]
Applied rustfmt fix for the failing `Rust packages (rs-sdk-ffi) / Formatting` check.\n\n- Commit: https://github.com/dashpay/platform/pull/3131/commits/63971b55f3752877f6d9a33b32ce0d290e857f0b\n- Touched files:\n  - `packages/rs-sdk-ffi/src/address_sync/mod.rs`\n  - `packages/rs-sdk-ffi/src/address_sync/provider.rs`\n\nCI has been retriggered on this push.

File: packages/rs-sdk/tests/fetch/generated_data.rs [hunk 1]
```diff
@@ -24,7 +24,9 @@ pub const IDENTITY_ID_2: Identifier = Identifier::new([2; 32]);
 /// Third identity used in the data contract above created by init chain for testing
 /// See `/packages/rs-drive-abci/src/execution/platform_events/initialization/create_genesis_state/mod.rs#L49`
 pub const IDENTITY_ID_3: Identifier = Identifier::new([3; 32]);
-/// Token ID that doesn't exist
+/// Token ID that doesn't exist in test state.
+/// NOTE: This has the same byte pattern as IDENTITY_ID_1 ([1; 32]). This is acceptable because
+/// token IDs and identity IDs occupy different namespaces in the state tree.
```
Comment (PastaPastaPasta) [inline packages/rs-sdk/tests/fetch/generated_data.rs]: unrelated changes?

File: packages/rs-sdk/tests/fetch/identity.rs [hunk 2]
```diff
@@ -189,7 +224,9 @@ async fn test_identity_public_keys_specific_read() {
 /// Given some non-unique public key, when I fetch identity that uses this key, I get associated identities containing this key
 #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
 async fn test_fetch_identity_by_non_unique_public_keys() {
-    /// Unique key hash, generated when creating test data
+    /// Key hash generated during test data creation in `create_genesis_state`.
+    /// Used to test fetching identities by non-unique public key hash.
+    /// Source: packages/rs-drive-abci/src/execution/platform_events/initialization/create_genesis_state/mod.rs
```
Comment (PastaPastaPasta) [inline packages/rs-sdk/tests/fetch/identity.rs]: more comment changes? please review this pr, and minimize unneeded changes

**Review by QuantumExplorer — APPROVED** [review]
